elastic/kibana
# [Fleet] making service_token an output secret

## Summary

Related to https://github.com/elastic/kibana/issues/104986

Making remote ES output's service_token a secret.

fleet-server change here: https://github.com/elastic/fleet-server/pull/3051#discussion_r1406183654

Steps to verify:
 - Enable remote ES output and output secrets in `kibana.dev.yml` locally:
 ```
xpack.fleet.enableExperimental: ['remoteESOutput', 'outputSecretsStorage']
```
- Start es, kibana, fleet-server locally and start a second es locally
 - see detailed steps here: https://github.com/elastic/fleet-server/pull/3051
- Create a remote ES output, verify that the service_token is stored as a secret reference
```
GET .kibana_ingest/_search?q=type:ingest-outputs
```
- Verify that the enrolled agent sends data to the remote ES successfully

<img width="561" alt="image" src="https://github.com/elastic/kibana/assets/90178898/122d9800-a2ec-47f8-97a7-acf64b87172a">
<img width="549" alt="image" src="https://github.com/elastic/kibana/assets/90178898/e1751bdd-5aaf-4f68-9f92-7076b306cdfe">



### Checklist
